src/code/z_skin_awb.c

```diff
@@ -1,13 +1,204 @@
+    s32 limbCount;
+    s32 i;
+    SkinLimb** skeleton;
+    SkeletonHeader* virtSkelHeader = Lib_SegmentedToVirtual(skeletonHeader);
+
+    skin->limbCount = virtSkelHeader->limbCount;
+    skin->skeletonHeader = virtSkelHeader;
+    limbCount = skin->skeletonHeader->limbCount;
+
+    skeleton = Lib_SegmentedToVirtual(skin->skeletonHeader->segment);
+    skin->vtxTable = ZeldaArena_Malloc(limbCount * sizeof(SkinLimbVtx));
+
+    for (i = 0; i < limbCount; i++) {
+        SkinLimbVtx* vtxEntry = &skin->vtxTable[i];
+
+        if ((((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segmentType != SKIN_LIMB_TYPE_ANIMATED) ||
+            (((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segment == NULL)) {
+            vtxEntry->index = 0;
+
+            vtxEntry->buf[0] = NULL;
+            vtxEntry->buf[1] = NULL;
+        } else {
+            SkinAnimatedLimbData* animatedLimbData =
+                Lib_SegmentedToVirtual((((SkinLimb*)Lib_SegmentedToVirtual(skeleton[i]))->segment));
+
+            { s32 tmp; }
+
+            vtxEntry->index = 0;
+            vtxEntry->buf[0] = ZeldaArena_Malloc(animatedLimbData->totalVtxCount * sizeof(Vtx));
+            vtxEntry->buf[1] = ZeldaArena_Malloc(animatedLimbData->totalVtxCount * sizeof(Vtx));
+
+            Skin_InitAnimatedLimb(gameState, skin, i);
+        }
+    }
+
+    SkelAnime_InitSkin(gameState, &skin->skelAnime, skeletonHeader, animationHeader);
+}
+
+/**
+ * Frees the dynamically allocated Vtx and SkinLimbVtx buffers and tables
+ */
+void Skin_Free(GameState* gameState, Skin* skin) {
+    if (skin->vtxTable != NULL) {
+        s32 i;
+
+        for (i = 0; i < skin->limbCount; i++) {
+            if (skin->vtxTable[i].buf[0] != NULL) {
+                ZeldaArena_Free(skin->vtxTable[i].buf[0]);
+                skin->vtxTable[i].buf[0] = NULL;
+            }
+            if (skin->vtxTable[i].buf[1] != NULL) {
+                ZeldaArena_Free(skin->vtxTable[i].buf[1]);
+                skin->vtxTable[i].buf[1] = NULL;
+            }
+        }
+
+        if (skin->vtxTable != NULL) {
+            ZeldaArena_Free(skin->vtxTable);
+        }
+
+        SkelAnime_Free(&skin->skelAnime, (GlobalContext*)gameState);
+    }
+}
+
+s32 func_801387D4(Skin* skin, SkinLimb** skeleton, MtxF* limbMatrices, u8 parentIndex, u8 limbIndex) {
+    s32 pad;
+    SkinLimb* limb = Lib_SegmentedToVirtual(skeleton[limbIndex]);
+    MtxF* mtx;
+    s32 ret;
+    MtxF sp28;
+
+    if (parentIndex == LIMB_DONE) {
+        SkinMatrix_GetClear(&mtx);
+    } else {
+        mtx = &limbMatrices[(s32)parentIndex];
+    }
+
+    SkinMatrix_MtxFMtxFMult(mtx, &limbMatrices[limbIndex], &sp28);
+    SkinMatrix_MtxFCopy(&sp28, &limbMatrices[limbIndex]);
+
+    if (limb->child != LIMB_DONE) {
+        ret = func_801387D4(skin, skeleton, limbMatrices, limbIndex, limb->child);
+        if (ret) { // func_801387D4 only returns false
```

Comment:
```suggestion
        if (ret) { // func_801387D4 can only return false
```
and elsewhere.

Reply:
fwiw I removed these comments from the oot version-- I feel like the fact that it only returns false in practice doesnt matter for the intent of what the check is doing here. It doesnt feel important to me to single out